azurelinuxagent/ga/update.py

```diff
@@ -606,18 +519,18 @@ def _processing_new_extensions_goal_state(self):
         egs = self._goal_state.extensions_goal_state
         return self._goal_state is not None and egs.id != self._last_extensions_gs_id and not egs.is_outdated
 
-    def _process_goal_state(self, exthandlers_handler, remote_access_handler):
+    def _process_goal_state(self, exthandlers_handler, remote_access_handler, agent_update_handler):
         protocol = exthandlers_handler.protocol
 
         # update self._goal_state
         if not self._try_update_goal_state(protocol):
             # agent updates and status reporting should be done even when the goal state is not updated
-            self.__update_guest_agent(protocol)
-            self._report_status(exthandlers_handler)
+            agent_update_handler.run()
+            self._report_status(exthandlers_handler, agent_update_handler, gs_updated=False)
```

Comment:
it seems to me like _report_status should not take a dependency on the agent_update_handler... all it needs is the update status, right? if yes, it may be better to pass the update status instead of the whole update_handler